### PR TITLE
[python] Add implementation of the `MultiscaleImage` to `main`

### PR DESCRIPTION
**Pull request by jp-dark**
Add the Python implementation for the new `MultiscaleImage` class.

### CODE REVIEW

File: apis/python/src/tiledbsoma/_multiscale_image.py [hunk 1]
```diff
@@ -93,9 +153,109 @@ def create(
...
+        image_type = ""
+        for val in axis_types:
+            try:
+                image_type += axis_type_map[val]
+            except KeyError as ke:
+                raise ValueError("Invalid axis type name '{val}'.") from ke
+        schema = MultiscaleImageSchema(
+            ImageProperties(
+                name="reference_level",
+                image_type=image_type,
```
Comment (nguyenv) [inline apis/python/src/tiledbsoma/_multiscale_image.py:176]: nit: This avoids creating a new immutable string object each time the string is appended

```suggestion
        image_type = []
        for val in axis_types:
            try:
                image_type.append(axis_type_map[val])
            except KeyError as ke:
                raise ValueError("Invalid axis type name '{val}'.") from ke
        schema = MultiscaleImageSchema(
            ImageProperties(
                name="reference_level",
                image_type="".join(image_type),
```

**Review by johnkerl — APPROVED** [review]

File: apis/python/src/tiledbsoma/_multiscale_image.py [hunk 2]
```diff
@@ -252,4 +628,82 @@ def reference_level_properties(self) -> "ImageProperties":
...
+            "i": pa.int32(),
+            "l": pa.int64(),
+            "C": pa.uint8(),
+            "S": pa.uint16(),
+            "I": pa.uint32(),
+            "L": pa.uint64(),
+            "f": pa.float32(),
+            "g": pa.float64(),
+        }
+        type = _carrow_to_pyarrow[type_str]
```
Comment (nguyenv) [inline apis/python/src/tiledbsoma/_multiscale_image.py:708]: Should move into `_arrow_types.py` to reside along with `pyarrow_to_carrow_type`

File: apis/python/src/tiledbsoma/_spatial_util.py [hunk 3]
```diff
@@ -91,6 +91,48 @@ def transform_region(
...
+    coords: options.DenseNDCoords = []
+    for axis in image_type:
+        if axis == "C":
+            coords.append(channel_coords)  # type: ignore[attr-defined]
+        if axis == "X":
+            coords.append(slice(x_min, x_max))  # type: ignore[attr-defined]
+        if axis == "Y":
+            coords.append(slice(y_min, y_max))  # type: ignore[attr-defined]
+        if axis == "Z":
+            raise NotImplementedError()
```
Comment (nguyenv) [inline apis/python/src/tiledbsoma/_spatial_util.py:132]: Maybe a more specific comment about how 3D is not implemented yet, if that's possible to add in here

**Review by nguyenv — APPROVED** [review]
lgtm; just minor nits

File: apis/python/src/tiledbsoma/_multiscale_image.py (hunk 1 above)
Comment (jp-dark) [reply apis/python/src/tiledbsoma/_multiscale_image.py:176]: Nice.

**Pull request merged**